src/main/java/com/myFirstProject/myFirstProject/service/BasketServiceImpl.java

```diff
@@ -73,8 +74,8 @@ private PromoCode getPromoCodeFromDB(BasketReq basketReq) {
         return promoCode;
     }
 
-    private PromoCode getPromoCode(Long id) {
-        PromoCode promoCode = promoCodeRepository.findById(id);
+    private PromoCode getPromoCode(String id) {
+        PromoCode promoCode = promoCodeRepository.findById(id).get();
```

Comment:
possible NPE.

If promocode is absent - it throw NPE.

Golden rule for optional.get() using - you can call it only if you 100 % sure that it exist.

right code:
1) return optional 
2) use getOrThrow